libmeepgeom/meepgeom.cpp

```diff
@@ -416,19 +369,23 @@ static meep::realnum linear_interpolate(
 static void epsilon_file_material(material_data *md, vector3 p)
```

Comment:
Since you can now assign this material to an object, maybe `p` should be offset by the center of the object when computing ε.  (So the file is almost like a "texture" applied to the object?)

Reply:
At the moment, file-based materials are representations of entire geometries, not of individual objects---if two or more objects exist in the original geometry, they are all flattened into a single spatially-varying background. I think I can picture what you have in mind, but wouldn't this require a little more sophistication in the interface? For example, we would want to check that the material file specified for an object describes a grid whose dimensions are commensurate with the object's bounding box? If you have a specific application in mind we could think about how to define appropriate conventions.